core-bundle/src/Resources/contao/library/Contao/DcaLoader.php

```diff
@@ -198,6 +201,39 @@ private function addDefaultLabels($blnNoCache)
 			unset($v);
 		}
 	}
+
+	/**
+	 * Sets the parent table for the current table, if enabled and not set.
+	 */
+	private function setDynamicPTable(): void
+	{
+		if (!($GLOBALS['TL_DCA'][$this->strTable]['config']['dynamicPtable'] ?? false) || !empty($GLOBALS['TL_DCA'][$this->strTable]['config']['ptable']) || !isset($GLOBALS['BE_MOD']))
+		{
+			return;
+		}
+
+		if (null !== ($do = Input::get('do')))
+		{
+			foreach (array_merge(...array_values($GLOBALS['BE_MOD'])) as $key => $module)
+			{
+				if ($do === $key && isset($module['tables']) && \is_array($module['tables']))
```

Comment:
This should be an early return, too.

```suggestion
				if ($do !== $key || !isset($module['tables']) || !\is_array($module['tables']))
				{
					continue;
				}
```